frontend/device/kobo/powerd.lua

```diff
@@ -252,13 +252,22 @@ function KoboPowerD:setWarmth(warmth)
     if self.fl == nil then return end
     if not warmth and self.auto_warmth then
         self:calculateAutoWarmth()
+        self:stateChanged()
     end
     self.fl_warmth = warmth or self.fl_warmth
     -- Don't turn the light back on on legacy NaturalLight devices just for the sake of setting the warmth!
     -- That's because we can only set warmth independently of brightness on devices with a mixer.
     -- On older ones, calling setWarmth *will* actually set the brightness, too!
     if self.device:hasNaturalLightMixer() or self:isFrontlightOnHW() then
         self.fl:setWarmth(self.fl_warmth)
+        self:stateChanged()
+    end
+end
+
+function KoboPowerD:getWarmth()
+    if self.fl == nil then return end
+    if self.device:hasNaturalLightMixer() or self:isFrontlightOnHW() then
```

Comment:
This probably ought to be a simple `hasNaturalLight` check [when in doubt, check what the FrontLight widget does, it should handle that distinction properly].

Reply:
I trust you :) But it should be checked on a Kobo with and on an Kobo without naturalLight.

Reply:
Yeah, I'll have some time to test it next Thursday if nobody got to it first ;).